dev/src/pipeline.ts

```diff
@@ -484,14 +477,226 @@ export class Pipeline<AppModelType = firestore.DocumentData>
+   * }</pre>
+   *
+   * @param other The other {@code Pipeline} that is part of union.
+   * @return A new {@code Pipeline} object with this stage appended to the stage list.
+   */
+  union(
+    other: FirebaseFirestore.Pipeline<AppModelType>
+  ): FirebaseFirestore.Pipeline<AppModelType> {
+    return this._addStage(new Union(other));
+  }
+
+  /**
+   * Produces a document for each element in array found in previous stage document.
+   *
+   * <p>For each previous stage document, this stage will emit zero or more augmented documents. The
+   * input array found in the previous stage document field specified by the `fieldName` parameter,
+   * will for each input array element produce an augmented document. The input array element will
+   * augment the previous stage document by replacing the field specified by `fieldName` parameter
+   * with the element value.
+   *
+   * <p>In other words, the field containing the input array will be removed from the augmented
+   * document and replaced by the corresponding array element.
+   *
+   * <p>Example:
+   *
+   * <pre>{@code
+   * // Input:
+   * // { 'title': 'The Hitchhiker's Guide to the Galaxy', 'tags': [ 'comedy', 'space', 'adventure' ], ... }
+   *
+   * // Emit a book document for each tag of the book.
+   * firestore.pipeline().collection('books')
+   *     .unnest('tags');
+   *
+   * // Output:
+   * // { 'title': 'The Hitchhiker's Guide to the Galaxy', 'tags': 'comedy', ... }
+   * // { 'title': 'The Hitchhiker's Guide to the Galaxy', 'tags': 'space', ... }
+   * // { 'title': 'The Hitchhiker's Guide to the Galaxy', 'tags': 'adventure', ... }
+   * }</pre>
+   *
+   * @param field The name of the field containing the array.
+   * @return A new {@code Pipeline} object with this stage appended to the stage list.
+   */
+  unnest(
+    field: FirebaseFirestore.Selectable | string
+  ): FirebaseFirestore.Pipeline<AppModelType>;
+
+  /**
+   * Produces a document for each element in array found in previous stage document.
+   *
+   * <p>For each previous stage document, this stage will emit zero or more augmented documents. The
+   * input array found in the previous stage document field specified by the `fieldName` parameter,
+   * will for each input array element produce an augmented document. The input array element will
+   * augment the previous stage document by replacing the field specified by `fieldName` parameter
+   * with the element value.
+   *
+   * <p>In other words, the field containing the input array will be removed from the augmented
+   * document and replaced by the corresponding array element.
+   *
+   * <p>Example:
+   *
+   * <pre>{@code
+   * // Input:
+   * // { 'title': 'The Hitchhiker's Guide to the Galaxy', 'tags': [ 'comedy', 'space', 'adventure' ], ... }
+   *
+   * // Emit a book document for each tag of the book.
+   * firestore.pipeline().collection('books')
+   *     .unnest({ field: 'tags', UnnestOptions.indexField('tagIndex'));
```

Comment:
`UnnestOptions.indexField('tagIndex')` doesn't look like like it's implemented. Is this correct?